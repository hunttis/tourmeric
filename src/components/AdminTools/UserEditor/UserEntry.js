import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-localize-redux';

export const UserEntry = ({ openEditModal, openCreditModal, openDeleteModal, userData, creditAmount }) => (
  <div className="userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd">
    <div className="card">

      <div className="card-header">
        <div className="card-header-title">
          {userData.displayName} - {userData.email}
        </div>
        {userData.avatarUrl &&
        <div className="card-header-icon">
          <figure className="image is-48x48">
            <img src={userData.avatarUrl} alt="User avatar" />
          </figure>
        </div>
      }
      </div>

      <div className="card-content">
        <ul>
          {userData.displayName &&
            <li><Translate id="displayname" />: {userData.displayName}</li>
          }
          {userData.username &&
            <li><Translate id="username" />: {userData.username}</li>
          }
          <li><Translate id="email" />: {userData.email}</li>
          {userData.avatarUrl &&
            <li><Translate id="avatarurl" />:&nbsp;
              <a target="_blank" rel="noopener noreferrer" href={userData.avatarUrl}>{userData.avatarUrl.substr(0, 20)}...</a>
            </li>
          }
          {userData.dciNumber &&
            <li><Translate id="dcinumber" />: {userData.dciNumber}</li>
          }
          <li><Translate id="storecredit" />: <span className="has-text-success">{creditAmount} â‚¬</span></li>
        </ul>
      </div>
      <div className="card-footer">
        <div className="card-footer-item button is-primary" onClick={openEditModal}>
          <Translate id="edit" />
        </div>
        <div className="card-footer-item button is-info" onClick={openCreditModal}>
          <Translate id="storecredit" />
        </div>
        <div className="card-footer-item button is-danger" onClick={openDeleteModal}>
          <Translate id="delete" />
        </div>
      </div>
    </div>
  </div>
);

UserEntry.propTypes = {
  openEditModal: PropTypes.func,
  openDeleteModal: PropTypes.func,
  openCreditModal: PropTypes.func,
  userData: PropTypes.object,
  creditAmount: PropTypes.number,
};
