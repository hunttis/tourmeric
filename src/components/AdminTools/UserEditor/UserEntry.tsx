import React from 'react';
import { Translate } from 'react-localize-redux';

import CreditAmounts from './CreditAmounts-container';
import { User } from '~/models/ReduxState';

interface Props {
  openEditModal: () => void;
  openDisableModal: () => void;
  openCreditModal: () => void;
  userId: string;
  userData: User;
}

export const UserEntry = ({ openEditModal, openCreditModal, openDisableModal, userData, userId }: Props) => (
  <div className="userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd">
    <div className="card">

      <div className="card-header">
        <div className="card-header-title">
          {userData.firstName} {userData.lastName} - {userData.email}
        </div>
        {userData.avatarUrl &&
          <div className="card-header-icon">
            <figure className="image is-48x48">
              <img src={userData.avatarUrl} alt="User avatar" />
            </figure>
          </div>
        }
      </div>

      <div className="card-content">
        <ul>
          {userData.firstName &&
            <li><Translate id="firstname" />: {userData.firstName}</li>
          }
          {userData.lastName &&
            <li><Translate id="lastname" />: {userData.lastName}</li>
          }
          {userData.username &&
            <li><Translate id="username" />: {userData.username}</li>
          }
          <li><Translate id="email" />: {userData.email}</li>
          {userData.avatarUrl &&
            <li><Translate id="avatarurl" />:&nbsp;
              <a target="_blank" rel="noopener noreferrer" href={userData.avatarUrl}>{userData.avatarUrl.substr(0, 20)}...</a>
            </li>
          }
          {userData.dciNumber &&
            <li><Translate id="dcinumber" />: {userData.dciNumber}</li>
          }
          <li>&nbsp;</li>
          <li><strong><Translate id="storecredit" /></strong></li>
          <CreditAmounts userId={userId} />
        </ul>
      </div>
      <div className="card-footer">
        <div className="card-footer-item button is-primary" onClick={openEditModal}>
          <Translate id="edit" />
        </div>
        <div className="card-footer-item button is-info" onClick={openCreditModal}>
          <Translate id="storecredit" />
        </div>
        <div className="card-footer-item button is-danger" onClick={openDisableModal}>
          <Translate id="delete" />
        </div>
      </div>
    </div>
  </div>
);
