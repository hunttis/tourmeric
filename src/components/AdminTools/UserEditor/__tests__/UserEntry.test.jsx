import React from 'react';
import { shallow } from 'enzyme';
import { UserEntry } from '../UserEntry';

describe('UserEntry Tests', () => {

  it('Compares UserEntry to snapshot', () => {

    const mockUserData = {}; // Todo mock user data
    const mockOpenModal = () => {};
    const mockDeleteModal = () => {};

    const userEntry = shallow(<UserEntry
      userData={mockUserData}
      openEditModal={mockOpenModal}
      openDeleteModal={mockDeleteModal}
    />);

    expect(userEntry.exists()).toBe(true);
    expect(userEntry).toMatchSnapshot();

  });

  // TODO: Write tests for clicking the modals

});
