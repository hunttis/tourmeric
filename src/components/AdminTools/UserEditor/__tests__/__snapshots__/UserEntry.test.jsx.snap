// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEntry Tests Compares UserEntry to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserEntry
    openDeleteModal={[Function]}
    openEditModal={[Function]}
    userData={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card"
      >
        <div
          className="card-header"
        >
          <div
            className="card-header-title"
          >
             
             - 
          </div>
        </div>
        <div
          className="card-content"
        >
          <ul>
            <li>
              <Translate
                id="email"
              />
              : 
            </li>
            <li>
               
            </li>
            <li>
              <strong>
                <Translate
                  id="storecredit"
                />
              </strong>
            </li>
            <Connect(Connect(CreditAmounts)) />
          </ul>
        </div>
        <div
          className="card-footer"
        >
          <div
            className="card-footer-item button is-primary"
            onClick={[Function]}
          >
            <Translate
              id="edit"
            />
          </div>
          <div
            className="card-footer-item button is-info"
          >
            <Translate
              id="storecredit"
            />
          </div>
          <div
            className="card-footer-item button is-danger"
          >
            <Translate
              id="delete"
            />
          </div>
        </div>
      </div>,
      "className": "userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-header"
          >
            <div
              className="card-header-title"
            >
               
               - 
            </div>
          </div>,
          <div
            className="card-content"
          >
            <ul>
              <li>
                <Translate
                  id="email"
                />
                : 
              </li>
              <li>
                 
              </li>
              <li>
                <strong>
                  <Translate
                    id="storecredit"
                  />
                </strong>
              </li>
              <Connect(Connect(CreditAmounts)) />
            </ul>
          </div>,
          <div
            className="card-footer"
          >
            <div
              className="card-footer-item button is-primary"
              onClick={[Function]}
            >
              <Translate
                id="edit"
              />
            </div>
            <div
              className="card-footer-item button is-info"
            >
              <Translate
                id="storecredit"
              />
            </div>
            <div
              className="card-footer-item button is-danger"
            >
              <Translate
                id="delete"
              />
            </div>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-header-title"
              >
                 
                 - 
              </div>,
              undefined,
            ],
            "className": "card-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  " ",
                  undefined,
                  " - ",
                  undefined,
                ],
                "className": "card-header-title",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                " ",
                undefined,
                " - ",
                undefined,
              ],
              "type": "div",
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul>
              <li>
                <Translate
                  id="email"
                />
                : 
              </li>
              <li>
                 
              </li>
              <li>
                <strong>
                  <Translate
                    id="storecredit"
                  />
                </strong>
              </li>
              <Connect(Connect(CreditAmounts)) />
            </ul>,
            "className": "card-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                undefined,
                undefined,
                <li>
                  <Translate
                    id="email"
                  />
                  : 
                </li>,
                undefined,
                undefined,
                <li>
                   
                </li>,
                <li>
                  <strong>
                    <Translate
                      id="storecredit"
                    />
                  </strong>
                </li>,
                <Connect(Connect(CreditAmounts)) />,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              undefined,
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Translate
                      id="email"
                    />,
                    ": ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "email",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  ": ",
                  undefined,
                ],
                "type": "li",
              },
              undefined,
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " ",
                },
                "ref": null,
                "rendered": " ",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <strong>
                    <Translate
                      id="storecredit"
                    />
                  </strong>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Translate
                      id="storecredit"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "id": "storecredit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "strong",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "userId": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="card-footer-item button is-primary"
                onClick={[Function]}
              >
                <Translate
                  id="edit"
                />
              </div>,
              <div
                className="card-footer-item button is-info"
              >
                <Translate
                  id="storecredit"
                />
              </div>,
              <div
                className="card-footer-item button is-danger"
              >
                <Translate
                  id="delete"
                />
              </div>,
            ],
            "className": "card-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Translate
                  id="edit"
                />,
                "className": "card-footer-item button is-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "edit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Translate
                  id="storecredit"
                />,
                "className": "card-footer-item button is-info",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "storecredit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Translate
                  id="delete"
                />,
                "className": "card-footer-item button is-danger",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "id": "delete",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card"
        >
          <div
            className="card-header"
          >
            <div
              className="card-header-title"
            >
               
               - 
            </div>
          </div>
          <div
            className="card-content"
          >
            <ul>
              <li>
                <Translate
                  id="email"
                />
                : 
              </li>
              <li>
                 
              </li>
              <li>
                <strong>
                  <Translate
                    id="storecredit"
                  />
                </strong>
              </li>
              <Connect(Connect(CreditAmounts)) />
            </ul>
          </div>
          <div
            className="card-footer"
          >
            <div
              className="card-footer-item button is-primary"
              onClick={[Function]}
            >
              <Translate
                id="edit"
              />
            </div>
            <div
              className="card-footer-item button is-info"
            >
              <Translate
                id="storecredit"
              />
            </div>
            <div
              className="card-footer-item button is-danger"
            >
              <Translate
                id="delete"
              />
            </div>
          </div>
        </div>,
        "className": "userentry column is-half is-one-third-desktop is-one-third-widescreen is-one-quarter-fullhd",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="card-header"
            >
              <div
                className="card-header-title"
              >
                 
                 - 
              </div>
            </div>,
            <div
              className="card-content"
            >
              <ul>
                <li>
                  <Translate
                    id="email"
                  />
                  : 
                </li>
                <li>
                   
                </li>
                <li>
                  <strong>
                    <Translate
                      id="storecredit"
                    />
                  </strong>
                </li>
                <Connect(Connect(CreditAmounts)) />
              </ul>
            </div>,
            <div
              className="card-footer"
            >
              <div
                className="card-footer-item button is-primary"
                onClick={[Function]}
              >
                <Translate
                  id="edit"
                />
              </div>
              <div
                className="card-footer-item button is-info"
              >
                <Translate
                  id="storecredit"
                />
              </div>
              <div
                className="card-footer-item button is-danger"
              >
                <Translate
                  id="delete"
                />
              </div>
            </div>,
          ],
          "className": "card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-header-title"
                >
                   
                   - 
                </div>,
                undefined,
              ],
              "className": "card-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    undefined,
                    " - ",
                    undefined,
                  ],
                  "className": "card-header-title",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  undefined,
                  " - ",
                  undefined,
                ],
                "type": "div",
              },
              undefined,
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul>
                <li>
                  <Translate
                    id="email"
                  />
                  : 
                </li>
                <li>
                   
                </li>
                <li>
                  <strong>
                    <Translate
                      id="storecredit"
                    />
                  </strong>
                </li>
                <Connect(Connect(CreditAmounts)) />
              </ul>,
              "className": "card-content",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  undefined,
                  undefined,
                  <li>
                    <Translate
                      id="email"
                    />
                    : 
                  </li>,
                  undefined,
                  undefined,
                  <li>
                     
                  </li>,
                  <li>
                    <strong>
                      <Translate
                        id="storecredit"
                      />
                    </strong>
                  </li>,
                  <Connect(Connect(CreditAmounts)) />,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                undefined,
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Translate
                        id="email"
                      />,
                      ": ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "id": "email",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    ": ",
                    undefined,
                  ],
                  "type": "li",
                },
                undefined,
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " ",
                  },
                  "ref": null,
                  "rendered": " ",
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <strong>
                      <Translate
                        id="storecredit"
                      />
                    </strong>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Translate
                        id="storecredit"
                      />,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "id": "storecredit",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "strong",
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "userId": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="card-footer-item button is-primary"
                  onClick={[Function]}
                >
                  <Translate
                    id="edit"
                  />
                </div>,
                <div
                  className="card-footer-item button is-info"
                >
                  <Translate
                    id="storecredit"
                  />
                </div>,
                <div
                  className="card-footer-item button is-danger"
                >
                  <Translate
                    id="delete"
                  />
                </div>,
              ],
              "className": "card-footer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Translate
                    id="edit"
                  />,
                  "className": "card-footer-item button is-primary",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "edit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Translate
                    id="storecredit"
                  />,
                  "className": "card-footer-item button is-info",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "storecredit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Translate
                    id="delete"
                  />,
                  "className": "card-footer-item button is-danger",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "id": "delete",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
